src/mudlet.cpp

```diff
@@ -672,242 +376,613 @@ mudlet::mudlet()
+                translation.mNativeName = QStringLiteral("简化字");
+            } else if (!languageCode.compare(QLatin1String("zh_TW"), Qt::CaseInsensitive)) {
+                translation.mNativeName = QStringLiteral("繁體字");
+            } else if (!languageCode.compare(QLatin1String("nl_NL"), Qt::CaseInsensitive)) {
+                translation.mNativeName = QStringLiteral("Nederlands");
+            } else if (!languageCode.compare(QLatin1String("fr_FR"), Qt::CaseInsensitive)) {
+                translation.mNativeName = QStringLiteral("Français");
+            } else if (!languageCode.compare(QLatin1String("de_DE"), Qt::CaseInsensitive)) {
+                translation.mNativeName = QStringLiteral("Deutsch");
+            } else if (!languageCode.compare(QLatin1String("el_GR"), Qt::CaseInsensitive)) {
+                translation.mNativeName = QStringLiteral("ελληνικά");
+            } else if (!languageCode.compare(QLatin1String("it_IT"), Qt::CaseInsensitive)) {
+                translation.mNativeName = QStringLiteral("Italiano");
+            } else if (!languageCode.compare(QLatin1String("pl_PL"), Qt::CaseInsensitive)) {
+                translation.mNativeName = QStringLiteral("Polszczyzna");
+            } else if (!languageCode.compare(QLatin1String("ru_RU"), Qt::CaseInsensitive)) {
+                translation.mNativeName = QStringLiteral("Pусский");
+            } else if (!languageCode.compare(QLatin1String("es_ES"), Qt::CaseInsensitive)) {
+                translation.mNativeName = QStringLiteral("Español");
+            } else if (!languageCode.compare(QLatin1String("pt_PT"), Qt::CaseInsensitive)) {
+                translation.mNativeName = QStringLiteral("Portugês");
+            } else {
+                translation.mNativeName = languageCode;
+            }
+            translation.mMudletTranslationPathFileName = path % QLatin1Char('/') % translationFileName;
+            mTranslationsMap.insert(languageCode, translation);
+        } else {
+            // This is very unlikely to be reached as it means that a file that
+            // matched the naming to be a Mudlet translation file was not infact
+            // one...
+            qDebug().noquote().nospace() << "mudlet::scanForMudletTranslations(...) INFO - no Mudlet translation found for locale code:\"" << languageCode << "\".";
+        }
+        delete pMudletTranslator;
+    }
+}
+
+// To be used AFTER scanForMudletTranslations(...) has been called, this will
+// insert the corresponding Qt system translation pathFileNames (path to
+// filenames: qt_xx.qm or qt_xx_YY.qm) into the entries in the mTranslationsMap
+// QHash.
+void mudlet::scanForQtTranslations(const QString& path)
+{
+    mQtTranslationsPathName = path;
+    qDebug().nospace().noquote() << "mudlet::scanForQtTranslations(\"" << path << "\") INFO - Seeking Qt translations files in given location.";
+    QMutableHashIterator<QString, translation> itTranslation(mTranslationsMap);
```

Comment:
I just spotted https://github.com/qt/qtdeclarative/commit/1f01b44d20471a7f4a5029a4c0049e8296749fef - so let's avoid introducing them

Reply:
:-1: 

I like and am now quite familiar with this type of iterator (`Q`**qt-container-class-type**`Iterator`and though marginally slower than what I think you are suggesting they are more robust and less likely to blow up in unexpected ways when the contents of the container are ~~frobbed~~modified.

This meets my needs now - if someone wants to retrofit a different style of iterator in the future in all the places we currently use this style then they can if they really want to and they know what they are doing...

Reply:
Well, it's just better not to introduce functionality we know is on the horizon to be depreciated. Good to learn the new stuff.